python/foglamp/services/core/api/configuration.py

```diff
@@ -178,6 +181,75 @@
     return web.json_response(result)
 
 
+async def add_configuration_item(request):
+    """
+    Args:
+        request: A JSON object that defines the config item and has key-pair
+                 (default, type, description, value[optional])
+
+    Returns:
+        Json response with message key
+
+    :Example:
+        curl -d '{"default": "true", "description": "Test description", "type": "boolean"}' -X POST https://localhost:1995/foglamp/category/{new_config_item} --insecure
+        curl -d '{"default": "true", "description": "Test description", "type": "boolean", "value": "false"}' -X POST https://localhost:1995/foglamp/category/test_cat/{new_config_item} --insecure
+    """
+    category_name = request.match_info.get('category_name', None)
+    new_config_item = request.match_info.get('add_config_item', None)
+
+    try:
+        storage_client = connect.get_storage()
+        cf_mgr = ConfigurationManager(storage_client)
+
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a dictionary')
+
+        # if value key is in data then go ahead with data payload and validate
+        # else update the data payload with value key and set its value to default value and validate
+        val = data.get('value', None)
```

Comment:
what if default is also missing?

Reply:
value set to None and thereafter validate_category_val handles the default key like missing key with 400 http status code